ledger/src/blockstore_db.rs

```diff
@@ -303,13 +349,47 @@ impl Rocks {
                         db_options.set_max_open_files(-1);
 
                         Rocks(
-                            DB::open_cf_as_secondary(&db_options, path, &secondary_path, names)?,
+                            DB::open_cf_as_secondary(
+                                &db_options,
+                                path,
+                                &secondary_path,
+                                cf_names.clone(),
+                            )?,
                             ActualAccessType::Secondary,
+                            oldest_slot,
                         )
                     }
                 }
             }
         };
+        if db.1 == ActualAccessType::Primary {
+            // Setting ttl only makes sense for primary access mode; and forcing so with secondary
+            // access causes hard-errors from RocksDB...
+            for cf_name in cf_names {
+                if cf_name == TransactionStatusIndex::NAME {
+                    continue;
+                }
+
+                db.0.set_options_cf(
+                    db.cf_handle(cf_name),
+                    &[(
+                        "ttl",
+                        &std::env::var("SOLANA_ROCKSDB_COMPACTION_TTL")
+                            .unwrap_or(format!("{}", 60 * 60 * 24 * 3)),
+                    )],
+                )
+                .unwrap();
+                db.0.set_options_cf(
+                    db.cf_handle(cf_name),
+                    &[(
+                        "periodic_compaction_seconds",
+                        &std::env::var("SOLANA_ROCKSDB_PERIODIC_COMPACTION_SECONDS")
+                            .unwrap_or(format!("{}", 60 * 60 * 24 * 3)),
```

Comment:
1 day would be sensible value. and this value should be pegged to `DEFAULT_COMPACTION_SLOT_INTERVAL`.